### PR TITLE
ci: build vue in CI

### LINKED ISSUE #35: Build vue in CI
Incorporate a build step for vue in the ci.

### PR DESCRIPTION
**Pull request by mpfan**
This pr adds a build step to the ci to build vue. 

Some notes:
- The SPAController serves index.html for all request made to /app. This means when you are making the client make sure your path always begin with /app.
- All api endpoints should start with /api. 
- Api requests should be made to /api

For development make changes in the shopify-client. Do not create a template folder in src/main/resource.

**It's important to run spring first then the client. This is because the development proxy proxies request to the api to /api**

Closes #35

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+  - yarn
+  - yarn run build
+  - mkdir ../src/main/resources/templates
+  - mkdir ../src/main/resources/public
+  - cp ./dist/index.html ../src/main/resources/templates/
+  - cp -R ./dist/css ../src/main/resources/public/
+  - cp -R ./dist/img ../src/main/resources/public/
+  - cp -R ./dist/js ../src/main/resources/public/
+  - cd ..
+  - mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
```
Comment (Souheil-Yazji) [inline .travis.yml:18]: are you sure we need this?

File: pom.xml [hunk 2]
```diff
@@ -51,6 +46,10 @@
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-thymeleaf</artifactId>
```
Comment (Souheil-Yazji) [inline pom.xml:51]: Why is this here?

File: pom.xml [hunk 3]
```diff
@@ -28,11 +28,6 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-data-rest</artifactId>
         </dependency>
-        <dependency>
```
Comment (Souheil-Yazji) [inline pom.xml:31]: Postgres dependency not needed?

**Review by dershao — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (mpfan) [reply pom.xml:51]: need thymeleaf for the index.html (default view)

File: pom.xml (hunk 3 above)
Comment (mpfan) [reply pom.xml:31]: We had duplicated postgres dependencies. It was giving an error

Comment (Souheil-Yazji) [reply pom.xml:31]: oh yes I see line 51 now

**Review by Weebian — DISMISSED** [review]

**Review by Souheil-Yazji — APPROVED** [review]

**Review by dershao — APPROVED** [review]

**Review by Weebian — APPROVED** [review]
+1
